modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/EditNotificationTemplate.tsx

```diff
@@ -171,17 +173,22 @@ export default function EditNotificationTemplate({
 						objectDefinitionId,
 						subject,
 						to,
-					})
+					});
+					setTitle(name);
+				}
 			);
 		}
+		else {
+			setTitle(Liferay.Language.get('untitled-notification-template'));
+		}
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [notificationTemplateId]);
 
 	return (
 		<ClayForm onSubmit={handleSubmit}>
 			<ClayManagementToolbar className="lfr__notification-template-management-tollbar">
 				<ClayManagementToolbar.ItemList>
-					<h2>{Liferay.Language.get('notification-templates')}</h2>
+					<h2>{title}</h2>
```

Comment:
I have another idea for this bug, instead of creating a new state to put the name in, why not use the already defined `values.name`? We can just do a ternary condition to display the correct title and we wouldn't need the language key anymore like:

```tsx
{!values.name
	? Liferay.Language.get('notification-templates')
	: values.name}
```

Reply:
Hi @danvb112, I already tried this, but in this case, the title changes dynamically, and not only when the template is saved.

Reply:
I asked Julia Lira about this and she said that the title can't be changed dynamically because it isn't the pattern of the Objects.

Reply:
now i understand better thanks for explaining @MrJohn1911, I will review again